src/core/ext/client_channel/parse_address.c

```diff
@@ -49,9 +49,10 @@
 
 int parse_unix(grpc_uri *uri, grpc_resolved_address *resolved_addr) {
   struct sockaddr_un *un = (struct sockaddr_un *)resolved_addr->addr;
+  memset(un, 0, sizeof(*un));
 
   un->sun_family = AF_UNIX;
-  strcpy(un->sun_path, uri->path);
+  strncpy(un->sun_path, uri->path, sizeof(un->sun_path) - 1 /* null term'd */);
```

Comment:
strncpy() is notoriously difficult to use in a safe way, because if the string is longer than the specified length, it does not copy the trailing NUL byte to the destination string.  strlcpy() is a much safer alternative to strncpy(), but it's unfortunately not portable. :(

I assume that this misfeature of strncpy() is why you're using memset() above to initialize the whole struct.  However, that seems a bit fragile, because a future developer looking at this code might think "the memset() isn't really necessary, because we're explicitly setting all of the fields below".  An alternative would be to do something like this right after this line:

```
un->sun_path[sizeof(un->sun_path) - 1] = '\0';
```